compliance_checker/cf/cf.py

```diff
@@ -2729,28 +2730,62 @@ def check_cell_methods(self, ds):
+                                matches = regex.match(r'^\s*(?P<interval_number>\S+)\s+(?P<interval_units>\S+)\s*$', val)
+                                # attempt to get the number for the interval
+                                if not matches:
+                                    valid_modifier.messages.append('{}:cell_methods contains an interval specification that does not parse: "{}". Should be in format "interval: <number> <units>"'.format(var.name, val))
+                                else:
+                                    try:
+                                        float(matches.group('interval_number'))
+                                    except ValueError:
+                                        valid_modifier.messages.append('{}:cell_methods contains an interval value that does not parse as a numeric value: "{}".'.format(var.name, matches.group('interval_number')))
+                                    else:
+                                        valid_modifier.score += 1
+
+                                    # then the units
+                                    try:
+                                        Unit(matches.group('interval_units'))
+                                    except ValueError:
+                                        valid_modifier.messages.append('{}:cell_methods interval units "{}" is not parsable by UDUNITS.'.format(var.name, matches.group('interval_units')))
+                                    else:
+                                        valid_modifier.score += 1
+                            elif keyword == 'comment:':
+                                # comments can't really be invalid, except
+                                # if they come first or aren't last, and
+                                # maybe if they contain colons embedded in the
+                                # comment string
+                                valid_modifier.out_of += 1
+                                if len(split_vals) == 1:
+                                    valid_modifier.messages.append('If there is no standardized information, the keyword comment: should be omitted for variable {}'.format(var.name))
+                                # otherwise check that the comment is the last
+                                # item in the parentheses
+                                elif i != len(split_vals) - 1:
+                                    valid_modifier.messages.append('The non-standard "comment:" element must come after any standard elements in cell_methods for variable {}'.format(var.name))
+                                #
+                                else:
+                                    valid_modifier.score += 1
+                            else:
+                                valid_modifier.out_of += 1
+                                valid_modifier.messages.append('Invalid cell_methods keyword "{}" for variable {}. Must be one of [interval, comment]'.format(keyword, var.name))
```

Comment:
This is some awfully nested code, strongly recommend breaking the nested branches into functions.

Reply:
Will do.